src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1974,7 +1975,12 @@ public BuildData copyBuildData(Run build) {
         BuildData buildData = null;
         while (build != null) {
             List<BuildData> buildDataList = build.getActions(BuildData.class);
-            for (BuildData bd : buildDataList) {
+            // We need to get the latest recorded build data. It may happens that the build has more than one
+            // checkout of the same repo
+            // Not used yet to let the test fail and then we see how it works 
+            // List<BuildData> buildDataListReverted = reversedView(buildDataList); 
+            List<BuildData> buildDataListReverted = buildDataList;
```

Comment:
At this point, the reverted list is the same, so no change actually. Just to let the CI fail the new test. After that, we uncomment it and the test should pass

Reply:
Thanks for using this technique so that we can see the test fail, then have you make the change so that we can see it pass.